backend/src/core/models/managed-list-campaign.ts

```diff
@@ -0,0 +1,33 @@
+import {
+  Table,
+  Column,
+  Model,
+  DataType,
+  BelongsTo,
+  ForeignKey,
+} from 'sequelize-typescript'
+import { Campaign } from '@core/models/campaign'
+
+@Table({
+  tableName: 'managed_list_campaigns',
+  underscored: true,
+  timestamps: true,
+})
+export class ManagedListCampaign extends Model<ManagedListCampaign> {
+  @ForeignKey(() => Campaign)
+  @Column({
+    type: DataType.INTEGER,
+    allowNull: false,
+    primaryKey: true,
+  })
+  campaignId: number
+
+  @BelongsTo(() => Campaign)
+  campaign!: Campaign
+
+  @Column({
+    type: DataType.INTEGER,
+    allowNull: false,
+  })
+  managedListId!: number
```

Comment:
`!` not necessary.

[nit] since this is an association table, would make sense to use a composite primary key across both `campaignId` and `managedListId`? see [this](https://github.com/opengovsg/postmangovsg/pull/2093/files#diff-ddbd590b554e5ebb98f8cdc53cfa3dc2a6ba6ae011a79a19c3c62b93671f4468) for an example

Reply:
Yes `managedListId` is fetched from Phonebook API (existing).

>  how do we ensure that the user cannot arbitrarily associate their campaign with any manageListId?

That's a good point, thanks. I will add a BE check for the API by calling existing Phonebook API to check that the managedListId is indeed owned by the user.

As for the association table, yes one campaign can only be associated with at most one `managedListId`. One `managedListId` can be associated with multiple different campaigns. I think a composite primary key is not necessary since a primary key on campaignid would suffice. We could potentially implement this by the nullable column that you suggested. I tried to avoid that as I want to try to avoid making changes to the existing table as far as possible, lest it breaks something. Also, adding to a new table has it's advantages. When we search whether a campaign is using a `managedList`, the new table to search for is much smaller than the whole `campaigns` table, since it only consists of campaigns that are using `managedListId`.

Reply:
> When we search whether a campaign is using a managedList, the new table to search for is much smaller than the whole campaigns table, since it only consists of campaigns that are using managedListId.

> I will add a BE check for the API by calling existing Phonebook API to check that the managedListId is indeed owned by the user.

why I think an "association table" is a bad idea:
- you don't really need an association table for a one-to-one relationship mapping
- lookup speed should not matter, since the campaign table is indexed by `id` alr
- if you implement the BE check described, presumably. you'd need to pass the `userId` or the user's email to Phonebook API. if you store this in a a separate table, you would need to do another join with the campaign table anyway.

> I tried to avoid that as I want to try to avoid making changes to the existing table as far as possible, lest it breaks something.

I am pretty sure adding a nullable column to the campaign table won't break anything :)

Reply:
Have discussed this, we will stick to a new table for now